examples/notebooks/solvers/scikits-dae-solver.ipynb

```diff
@@ -0,0 +1,374 @@
+      "text/plain": [
+       "<Figure size 936x288 with 2 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "def exponential_decay_dae(t, y, ydot):\n",
+    "    return [-y[0] - ydot[0], 2 * y[0] - y[1]]\n",
+    "\n",
+    "# Solve\n",
+    "y0 = np.array([1, 2])\n",
+    "t_eval = np.linspace(0, 5, 20)\n",
+    "t_sol, y_sol = dae_solver.integrate(exponential_decay_dae, y0, t_eval)\n",
+    "\n",
+    "# Plot\n",
+    "t_fine = np.linspace(0,t_eval[-1],1000)\n",
+    "\n",
+    "def plot(t_sol, y_sol):    \n",
+    "    fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(13,4))\n",
+    "    ax1.plot(t_fine, np.exp(-t_fine), t_sol, y_sol[0], \"o\")\n",
+    "    ax1.set_xlabel(\"t\")\n",
+    "    ax1.legend([\"exp(-t)\", \"y_sol[0]\"], loc=\"best\")\n",
+    "\n",
+    "    ax2.plot(t_fine, 2*np.exp(-t_fine), t_sol, y_sol[1], \"o\")\n",
+    "    ax2.set_xlabel(\"t\")\n",
+    "    ax2.legend([\"2*exp(-t)\", \"y_sol[1]\"], loc=\"best\")\n",
+    "\n",
+    "    plt.tight_layout()\n",
+    "    plt.show()\n",
+    "    \n",
+    "plot(t_sol, y_sol)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can also provide the mass matrix and Jacobian (both must be provided together)"
```

Comment:
Maybe it is worth commenting here that here the equations are in the form F(t,y,ydot) = f(t,y) - M*ydot = 0, and that here mass matrix is M and jacobian in df/dy. If people are familiar with the SUNDIALS interface, the IDA defines the jacobian as dF/dy (and uses the chain rule) so doesn't make the distinction between the mass matrix and the jacobian (df/dy) like we do. As I'm writing this I think it's probably ok as you have it. The additional comment may cause more confusion.

Reply:
haha ok will leave as is